src/components/list/list.ts

```diff
@@ -0,0 +1,56 @@
+import {Component, ViewEncapsulation, ContentChildren, ContentChild, QueryList, Directive,
+  ElementRef, Renderer} from 'angular2/core';
+import {CONST_EXPR} from 'angular2/src/facade/lang';
+
+@Component({
+  selector: 'md-list',
+  host: {'role': 'list'},
+  template: '<ng-content></ng-content>',
+  styleUrls: ['./components/list/list.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdList {}
+
+@Directive({ selector: '[md-line]' })
+export class MdLine {}
+
+@Directive({ selector: '[md-list-avatar]' })
+export class MdListAvatar {}
+
+@Component({
+  selector: 'md-list-item',
+  host: {'role': 'listitem'},
+  templateUrl: './components/list/list-item.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class MdListItem {
+  @ContentChildren(MdLine)
+  set _lines(lines: QueryList<MdLine>) {
```

Comment:
I suspect this setter is actually not going to be called when the number of lines change. This should work, though:
```ts
_lines.changes.subscribe(_ => console.log(items.length));
```